ddtrace/contrib/dbapi/__init__.py

```diff
@@ -164,12 +165,30 @@ def fetchmany(self, *args, **kwargs):
         )
 
 
-def _get_config(new_cfg):
+# TODO: Remove once config.dbapi2 has been removed
+class _OverrideAttrDict(wrapt.ObjectProxy):
+    __slots__ = ("override", "base")
+    sentinel = object()
+
+    def __init__(self, override, base):
+        super(_OverrideAttrDict, self).__init__(override)
+        self.override = override
+        self.base = base
+
+    def __getattr__(self, name):
+        value = self.override.get(name, self.sentinel)
+        return getattr(self.base, name) if value == self.sentinel else value
+
+    def __getitem__(self, name):
+        value = self.override.get(name, self.sentinel)
+        return self.base.__getitem__(name) if value == self.sentinel else value
+
+
```

Comment:
Need to add

~~~ python

    def __bool__(self):
        return bool(self.override or self.base)

    def __contains__(self, item):
        return item in self.override or item in self.base
~~~

Reply:
@majorgreys  I think the `__bool__` must be implemented too

Reply:
Wondering what your reasoning was for this. Should we still add it in?

Reply:
It made the test for dbapi pass with my experiments so I thought it was needed. It looks like all tests are now passing without it so there's no need for it I suppose.